ztools/x509/json.go

```diff
@@ -228,37 +229,41 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 	jc.SubjectKeyInfo.KeyAlgorithm = c.PublicKeyAlgorithm
 
 	// Include all subject names, DNS names there are
-	for _, obj := range c.Subject.Names {
-
-		switch name := obj.Value.(type) {
-		case string:
-
-			flag := false
-
-			if len(name) > 2 && name[0] == '*' {
-				flag = govalidator.IsURL(name[2:])
-			} else {
-				flag = govalidator.IsURL(name)
-			}
+	name := c.Subject.CommonName
+	isValid := false
+
+	if len(name) > 2 && name[0] == '*' {
+		isValid = govalidator.IsURL(name[2:])
+	} else if len(name) > 2 && name[0] == '?' {
+		isValid = isValid || govalidator.IsURL(name[2:])
+	} else if !strings.Contains(name, ".") {
```

Comment:
It won't, because of line 232 in this commit: https://github.com/zmap/zgrab/commit/adccb4b0c2d79ad4d171ecbcc9f7921d2631284f#diff-b69ed0926a8809fc22de87e608774fe9R232